src/main/java/de/vill/conversion/ConvertFeatureCardinality.java

```diff
@@ -71,22 +73,77 @@ private void removeFeatureCardinality(Feature feature, FeatureModel featureModel
                 constraintReplacementMap.remove(feature.getFeatureName());
                 for (Constraint constraint : constraintsToClone) {
                     Constraint newConstraint = constraint.clone();
+                    if (newConstraint instanceof LiteralConstraint) {
+                        String toReplace = ((LiteralConstraint) newConstraint).getReference().getIdentifier();
+                        if (constraintReplacementMap.containsKey(toReplace)) {
+                            LiteralConstraint newLiteral = new LiteralConstraint(constraintReplacementMap.get(toReplace));
+                            LiteralConstraint subTreeRootConstraint = new LiteralConstraint(newChild);
+                            newConstraint = new ImplicationConstraint(subTreeRootConstraint, new ParenthesisConstraint(newLiteral));;
+                        }
+                    }else{
+                        adaptConstraint(subTreeClone, newConstraint, constraintReplacementMap);
+                        LiteralConstraint subTreeRootConstraint = new LiteralConstraint(newChild);
+                        newConstraint = new ImplicationConstraint(subTreeRootConstraint, new ParenthesisConstraint(newConstraint));
+                    }
                     featureModel.getOwnConstraints().add(newConstraint);
-                    adaptConstraint(subTreeClone, newConstraint, constraintReplacementMap);
                 }
             }
         }
+/*
+additional constraint with all cloned versions ored
```

Comment:
Do we need this commented-out fragmnet?

Reply:
This code was used to create an additional constraint, where are cloned features are or-ed. For a constraint that contains a cloned feature, it can then be replaced by this expression. We discussed this kind of constraint while talking about the semantic of constraints regarding feature cardinality. However, I did not include this kind of constraint and therefore commented it out again. I did not remove it, in case we want to use it later on.